test/cluster/test_multidc.py

```diff
@@ -143,3 +146,124 @@ async def test_query_dc_with_rf_0_does_not_crash_db(request: pytest.FixtureReque
         f"Expected {expected} from {select_query.query_string}, but got {first_node_results}"
     assert second_node_result is None, \
         f"Expected no results from {select_query.query_string}, but got {second_node_result}"
+
+@pytest.mark.asyncio
+async def test_create_and_alter_keyspace_with_altering_rf_and_racks(manager: ManagerClient):
+    """
+    This test verifies that creating and altering a keyspace keeps it RF-rack-valid.
+    If an operation would make it RF-rack-invalid, it should fail.
+    We can add a new rack or a data center and the existing keyspaces must still work fine.
+    """
+
+    cql = None
+    cfg = {"rf_rack_valid_keyspaces": "true"}
+
+    async def create_ok(rfs: List[int]) -> str:
+        ks = unique_name()
+        dcs = ", ".join([f"'dc{i + 1}': {rf}" for i, rf in enumerate(rfs)])
+        await cql.run_async(f"CREATE KEYSPACE {ks} WITH REPLICATION = {{'class': 'NetworkTopologyStrategy', {dcs}}} AND tablets = {{'enabled': true}}")
+        return ks
+
+    async def create_fail(rfs: List[int], failed_dc: int, rack_count: int) -> None:
+        ks = unique_name()
+        dcs = ", ".join([f"'dc{i + 1}': {rf}" for i, rf in enumerate(rfs)])
+
+        dc = f"dc{failed_dc}"
+        rf = rfs[failed_dc - 1]
+
+        err = r"The option `rf_rack_valid_keyspaces` is enabled. It requires that all keyspaces are RF-rack-valid. " \
+              f"That condition is violated: keyspace '{ks}' doesn't satisfy it for DC '{dc}': RF={rf} vs. rack count={rack_count}."
+
+        with pytest.raises(InvalidRequest, match=err):
+            await cql.run_async(f"CREATE KEYSPACE {ks} WITH REPLICATION = {{'class': 'NetworkTopologyStrategy', {dcs}}} AND tablets = {{'enabled': true}}")
+
+    async def alter_ok(ks: str, rfs: List[int]) -> None:
+        dcs = ", ".join([f"'dc{i + 1}': {rf}" for i, rf in enumerate(rfs)])
+        await cql.run_async(f"ALTER KEYSPACE {ks} WITH REPLICATION = {{'class': 'NetworkTopologyStrategy', {dcs}}}")
+
+    async def alter_fail(ks: str, rfs: List[int], failed_dc: int, rack_count: int) -> None:
+        dcs = ", ".join([f"'dc{i + 1}': {rf}" for i, rf in enumerate(rfs)])
+
+        dc = f"dc{failed_dc}"
+        rf = rfs[failed_dc - 1]
+
+        err = r"The option `rf_rack_valid_keyspaces` is enabled. It requires that all keyspaces are RF-rack-valid. " \
+              f"That condition is violated: keyspace '{ks}' doesn't satisfy it for DC '{dc}': RF={rf} vs. rack count={rack_count}."
+
+        with pytest.raises(InvalidRequest, match=err):
+            await cql.run_async(f"ALTER KEYSPACE {ks} WITH REPLICATION = {{'class': 'NetworkTopologyStrategy', {dcs}}}")
+
+    # dc1: r1=1, rzerotoken=1 (Note: zero-token nodes have NO impact. In this case, this rack should behave as if it never existed).
```

Comment:
It looks like we don't have tests for arbiter DCs (DCs with zero-token nodes only).